src/main/groovy/com/rundeck/plugins/ansible/ansible/AnsibleVault.java

```diff
@@ -71,58 +84,122 @@ public String encryptVariable(String key,
             System.out.println("encryptVariable " + key + ": " + procArgs);
         }
 
-        //send values to STDIN in order
-        List<String> stdinVariables = new ArrayList<>();
-        stdinVariables.add(content);
+        File promptFile = File.createTempFile("vault-prompt", ".log");
 
         Map<String, String> env = new HashMap<>();
-        env.put("VAULT_ID_SECRET", masterPassword);
+        env.put("LOG_PATH", promptFile.getAbsolutePath());
 
         Process proc = null;
 
         try {
-            proc = ProcessExecutor.builder().procArgs(procArgs)
+            proc = processExecutorBuilder.procArgs(procArgs)
                     .baseDirectory(baseDirectory.toFile())
-                    .stdinVariables(stdinVariables)
-                    .redirectErrorStream(true)
                     .environmentVariables(env)
+                    .redirectErrorStream(true)
                     .build().run();
 
-            StringBuilder stringBuilder = new StringBuilder();
-
-            final InputStream stdoutInputStream = proc.getInputStream();
-            final BufferedReader stdoutReader = new BufferedReader(new InputStreamReader(stdoutInputStream));
-
-            String line1 = null;
-            boolean capture = false;
-            while ((line1 = stdoutReader.readLine()) != null) {
-                if (line1.toLowerCase().contains("!vault")) {
-                    capture = true;
-                }
-                if (capture) {
-                    stringBuilder.append(line1).append("\n");
+            final InputStream proccesInputStream = proc.getInputStream();
+            final OutputStream processOutputStream = proc.getOutputStream();
+
+            //capture output thread
+            Callable<String> readOutputTask = () -> {
+                return readOutput(proccesInputStream);
+            };
+
+            ExecutorService executor = Executors.newSingleThreadExecutor();
+            Future<String> future = executor.submit(readOutputTask);
+
+            Thread stdinThread = new Thread(() -> sendValuesStdin(processOutputStream, masterPassword, content));
+
+            long start = System.currentTimeMillis();
+            long end = start + 60 * 1000;
+
+            //wait for prompt
+            boolean promptFound = false;
+            while (!promptFound && System.currentTimeMillis() < end) {
+                BufferedReader reader = new BufferedReader(new FileReader(promptFile));
+                String currentLine = reader.readLine();
+                if(currentLine!=null && currentLine.contains("Enter Password:")){
+                    promptFound = true;
+                    //send password / content
+                    stdinThread.start();
+                    reader.close();
+                }else{
+                    Thread.sleep(1500);
```

Comment:
Looks like `encryptVariable` is called in a serial loop for each extra var. With these 1.5 sec sleeps and a decent number of extra vars to encrypt, these steps may spend quite some time encrypting before it gets to the actual running.